File: WebApi/Models/TitleDto.ts
import type { ContactDto } from './ContactDto';

export interface TitleDto {
  id: number;
  titleName: string;
  createdDate: string;
  updatedDate?: string;
  contacts?: ContactDto[];
}

export interface CreateTitleDto {
  titleName: string;
}

export interface UpdateTitleDto {
  titleName: string;
}

File: DTOs/CustomerDto/TitleDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace cms_webapi.DTOs
{
    public class TitleDto
    {
        public int Id { get; set; }
        public string TitleName { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public List<ContactDto>? Contacts { get; set; }
    }

    public class CreateTitleDto
    {
        [Required]
        [MaxLength(50)]
        public string TitleName { get; set; } = string.Empty;
    }

    public class UpdateTitleDto
    {
        [Required]
        [MaxLength(50)]
        public string TitleName { get; set; } = string.Empty;
    }
}
